Add ReadOnly option to volumeMount in Pods

irumaru opened the pull request:
Added ReadOnly to the volumeMount option in Pods to support mounting of ReadOnlyMany(ROX) PVC.
https://github.com/grafana/k6-operator/issues/279

The default is ReadOnly: false.
The reason was to avoid destructive changes, considering the possibility that some users may already be writing to the volume in the init container.


Here are the results of a simple test.
[files.zip](https://github.com/user-attachments/files/15799552/files.zip)

Test Contents
- kubectl apply -f pvc.yaml to create pods to add pods and scenarios
- kubectl exec -it vp -- /bin/bash to add main.js to /mnt/vp in pods
- Create TestRun with kubectl apply -f main.yaml

result
describe-log.txt and get-log.txt are the execution results.
You can confirm that ReadOnly for k6-test-volume is true.

yorugac commented on pkg/types/script.go:
```diff
@@ -10,6 +10,7 @@ import (
 // Internal type created to support Spec.script options
 type Script struct {
 	Name     string // Name of ConfigMap or VolumeClaim or "LocalFile"
+	ReadOnly bool
```
```suggestion
	ReadOnly bool // VolumeClaim only
```

yorugac reviewed (changes requested):
@irumaru, sorry for the delay!
Thanks for working on this :raised_hands: PR LGTM: there's just a minor suggestion to improve comments. And one more thing: could you please create a folder in `config/samples` with code files from your archive?

irumaru commented:
@yorugac , Thanks for the review.

We have made the following two corrections.
- Added `config/samples/k6_v1alpha1_k6_with_readOnlyVolumeClaim.yaml`.
- Added comment to `pkg/types/script.go` as suggested.

Please review.

yorugac reviewed (approved):
Thank you for the update @irumaru! :raised_hands:
